features/lorawan/LoRaWANInterface.cpp

```diff
@@ -170,37 +110,22 @@ lorawan_status_t LoRaWANInterface::remove_channel_plan()
 int16_t LoRaWANInterface::send(uint8_t port, const uint8_t* data,
                                uint16_t length, int flags)
 {
-    if (_link_check_requested) {
-        // add a link check request with normal data, until the application
-        // explicitly removes it.
-        add_link_check_request();
-    }
+    return stk_obj().handle_tx(port, data, length, flags);
 
-    if (data) {
-        return stk_obj().handle_tx(port, data, length, flags);
-    } else {
-        return LORAWAN_STATUS_PARAMETER_INVALID;
-    }
 }
 
 int16_t LoRaWANInterface::receive(uint8_t port, uint8_t* data, uint16_t length,
                                   int flags)
 {
-    if (data && length > 0) {
-        return stk_obj().handle_rx(port, data, length, flags);
-    } else {
-        return LORAWAN_STATUS_PARAMETER_INVALID;
-    }
+    return stk_obj().handle_rx(port, data, length, flags);
 }
 
 lorawan_status_t LoRaWANInterface::add_app_callbacks(lorawan_app_callbacks_t *callbacks)
-  {
-
-     if (!callbacks || !callbacks->events) {
-         // Event Callback is mandatory
-         return LORAWAN_STATUS_PARAMETER_INVALID;
-     }
+{
+    return stk_obj().set_lora_callbacks(callbacks);
+}
 
-     stk_obj().set_lora_callbacks(callbacks);
-     return LORAWAN_STATUS_OK;
-  }
+lorawan_status_t LoRaWANInterface::set_device_class(const device_class_t device_class)
```

Comment:
(You could keep the `const` here though, meaning "I don't change `device_class` after function entry - just doesn't need to be in the prototype).